src/main/java/net/neoforged/neoforge/event/entity/living/ArmorHurtEvent.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright (c) NeoForged and contributors
+ * SPDX-License-Identifier: LGPL-2.1-only
+ */
+
+package net.neoforged.neoforge.event.entity.living;
+
+import java.util.Map;
+import net.minecraft.world.damagesource.DamageSource;
+import net.minecraft.world.entity.EquipmentSlot;
+import net.minecraft.world.entity.player.Player;
+import net.minecraft.world.item.ItemStack;
+import net.neoforged.bus.api.ICancellableEvent;
+import net.neoforged.neoforge.event.entity.player.PlayerEvent;
+import org.jetbrains.annotations.ApiStatus;
+
+/**
+ * Fired when a {@link Player}'s armor is dealt damage in {@link Player#actuallyHurt(DamageSource, float) actuallyHurt}.
+ * <p>
+ * This event is {@link ICancellableEvent cancelable}. Cancelling this event will ignore all damage modifications
+ * and result in the original damage being applied to the armor item.
+ * <p>
+ * This event does not have a result.
+ * <p>
+ * This event is fired on the {@link NeoForge#EVENT_BUS}
+ */
+public class ArmorHurtEvent extends PlayerEvent implements ICancellableEvent {
+    private final Map<EquipmentSlot, ItemStack> armorItemStack;
+    private final Map<EquipmentSlot, Float> originalDamage;
+    private final Map<EquipmentSlot, Float> damage;
```

Comment:
Any reason to not use 1 map containing an object with the stack, original damage, and new damage?

Reply:
Mostly just makes it easier to iterate, rather than needing to loop through the slots and call 3 different getters and a setter.

Reply:
updated.